### ISSUE TITLE
[css-fonts] Pseudo-generic font families

### ISSUE THREAD

**Issue by Crissov** [w3c/csswg-drafts#5065]
Several actual (Western) typeface names are sometimes used generically, in that there are either metrically compatible clones (with more liberal licenses) or multiple digitizations of the same lead-era typeface. 

Should implementations explicitly be allowed to treat them more like keywords in `<'font-family'>` if they are not quoted, i.e. choose an alternative font that is available? 

This would primarily affect the set of Microsoftʼs [Core fonts for the Web](https://en.wikipedia.org/wiki/Core_fonts_for_the_Web), which also contain the de facto standard mappings for the generic font families `serif` (_Times_ or _Times New Roman_), `sans-serif` (_Arial_) and `monospace` (_Courier_ or _Courier New_).

**Comment by Myndex** [comment]
It would be good to have a set of accessible fonts as defined standards, so long as Microsoft is not allowed to create them.

Courier New is one of the worst possible fonts for web use, next is Arial Condensed. Both are MS.

FWIW, I have a preprint of a paper on accessible fonts:

https://www.researchgate.net/publication/338149302_Evaluating_Fonts_Font_Family_Selection_for_Accessibility_Display_Readability

**Comment by Crissov** [comment]
See #5819, also #4910, #4566.

**Comment by PeterCon** [comment]
@Myndex: It's regrettable that, of all the fonts developed by Microsoft, you make a sweeping judgment based on Arial Narrow and Courier New, which were based on some earlier font designs and not specifically for optimizing on-screen legibility, while disregarding Georgia and Verdana, both of which were designed with on-screen legibility in mind and which you evaluate as good for accessibility and readability. Or Sitka Text, which was developed using scientific evaluations of legibility (see [Larson & Carter, 2016](https://www.microsoft.com/en-us/research/publication/sitka-a-collaboration-between-type-design-and-science/). 

Also, it would have been helpful if your paper on evaluating fonts for legibility had provided some details, or even reference to some separate work, explaining how you actually arrived at objectively-measured evaluations and what the scientific basis was for the evaluations.

**Comment by Myndex** [comment]
Hello Peter @PeterCon 

First to be clear, in the paper I never mentioned Microsoft. The paper was clearly stated as an _informal guidance_ on font characteristics for accessibility, and I gave **positive reviews to both Georgia and Verdana** (both of which were designed by noted type designer Matthew Carter.) _I certainly did not "disregard them" as you say._

Because it was an informal guidance booklet it does not come with a formal bibliography, but the in-progress, in depth work shall, as it's part of a larger work on contrast and readability for web based content. 

## _Accessibility_
Let me pause for a moment here: I am not talking about fonts as far as their aesthetic design. I am talking strictly about _accessibility_, including research on dyslexia, cognitive, contrast perception, and visual impairments. 

### Some accessibility points are, does a given font:
- have adequate tracking, kerning, and leading? (letter spacing and linespacing)
- have an adequate x-height? (Ideally ~60% of the font-size)
- have adequate weight? (Adequate stroke thickness at 400)
- have a "good" Cap-height? (For differentiating from x and allowing good leading)
- avoid common homoglyphs? (Meaning **rn** _vs_ **m** and **1** _vs_ **l** _vs_ **I** are distinct.)
- avoid mirror image glyphs? (Meaning **db** or **qp** or **MW** are not literal mirror copies)
- have good hinting for screen use? (Meaning is designed to render well to standard resolution screens.)

Most of these characteristics are discussed prominently in the current research into readability. For cites, I refer to Lovie-Kitchin, Bailey, Arditi, Legge all of whom I rely on and who have done the bulk of the work in readability for normal and impaired vision. All that said:

## Re: MS Fonts
Interestingly, Sitka Text was also a Matthew Carter creation, so not surprising it looks good. Nice to see.

As a result of your post, I did a much deeper dive into the history of "MS Fonts" by which I mean the "core web fonts" which MS distributed once upon a time. Those were the proprietary fonts _Andalé Mono, Arial, Arial Black, Comic Sans MS, Courier New, Georgia, Impact, Times New Roman, Trebuchet MS, Verdana_ and _Webdings,_

Of _THAT_ group, only Georgia, Verdana, and _maybe_ Trebuchet MS are "accessible friendly". The problem with Trebuchet is the default track/kerning is too tight, so it's not normally on my list of "good" fonts for accessibility.

But also of that group, Courier New is nothing short of an **unusable illegitimate version of the classic Courier typeface.** Arial is nothing but a pure copy of Helvetica, which I also give somewhat low marks to regarding _accessibility_. Times New Roman is just a pure copy of Times, and the small size and small x-height make it less than desirable. _Side Note: (The straight up copies was because Microsoft did not want to pay the license for the originals)._

As a result of the deeper dive, I see that _some_ of the fonts I loath the most were Microsoft commissions to Monotype. Hmmm. And looking further, Monotype is behind a few other fonts I consider "bad" with nothing to do with Microsoft... and or that are just knockoffs of some other font. Microsoft seems to have a close business relationship to Monotype. Hmmm.

Nevertheless, it was MS that _chose_ the core fonts, and including Courier New, Times New Roman, Arial, which are copies of Courier, Times, and Helvetica, respectively, because Microsoft did not want to pay the license fees for the original versions. Microsoft has more recent fonts that are also poor for accessibility. For that matter, most boundaries do considering that accessibility is not usually at the forefront of design, sad as that may be.

But getting back to _Courier New_ it is not even a faithful copy and it's horrendous for screen use. Examples:

<img width="598" alt="Screen Shot 2021-09-28 at 10 53 15 PM" src="https://user-images.githubusercontent.com/42009457/135211088-8c3a10e4-5149-4c3f-ab65-cecd4a9d2a9f.png">

_Real_ Courier is a fairly readable font, but the MS commissioned _Courier New_ is so very light it is unusable.

And I included some more recent MS examples, like Gulim, Cordia, and Iris ... like, **_wut?_**

### _And for the record, my critique is only and exclusively directed at accessibility issues of the cited fonts. If it's any consolation to you, many Google fonts are also anti-accessible._

And I do give MS credit for commissioning Matthew Carter to add some good solid designs to MS's fonts.

Thank you,

Andy

**Comment by Myndex** [comment]
Hi @Crissov 

Thinking about this and accessibility issues: I do think it would be good to specify some very standard fallbacks, but those fallback fonts should be ones that are accessible.

While writing a response to Peter's comment, it occurred to me that the license for the original core fonts might permit making modified versions for accessibility (I'd need to read the license carefully)... but if it is okay, then we could generate accessible versions, and those would be the defaults...


A

**Comment by PeterCon** [comment]
@Myndex Hi, Andy

Thanks for the clarifications.

>For the record, when I made that offhand remark in the post above, it was more focused toward the reviled "core web fonts".

That's not what your remark appears to be saying, though:

> It would be good to have a set of accessible fonts as defined standards, so long as Microsoft is not allowed to create them.

That's a general statement about any fonts that Microsoft has made or might ever make.

**Comment by Myndex** [comment]
Hi Peter @PeterCon 

I may have been grumpier than usual that day. And to be honest, I'm glad you called me on it — when someone does that my usual reaction is to look deeper into something. As a result, I ended up reading the paper on the research surrounding Sitka Text and I'm glad to know there is at least a part of MS that is making a proactive difference.

Nevertheless, who at MS was behind _Gulim, Cordia,_ and _Iris_? And what can MS do to fix the decades old problem of _Courier New_? From the standpoint of accessibility, these are some of the problems that formed negative opinion in this regard.

Thank you,

Andy

**Comment by astearns** [comment]
@PeterCon @Myndex I do not think this back-and-forth is contributing to the issue. But I am also a bit at a loss as to why this issue exists. 

@Crissov couldn't this have been a comment in https://github.com/w3c/csswg-drafts/issues/4910? 

@Myndex could you add a comment to https://github.com/w3c/csswg-drafts/issues/4910 with your suggestion about accessible fonts? 

Then we could close this out in favor of the more general ongoing discussion there about whether more generic font families is a good idea.

**Comment by Myndex** [comment]
Sorry Alan @astearns, moving my main comment over there...

**Comment by Crissov** [comment]
While this issue is somewhat related to the criteria for generic font families, it deals with a distinct, specific question:  
May implementations treat unquoted font families _of their own choosing_ like keywords?
